node/src/components/storage.rs

```diff
@@ -556,7 +556,25 @@ impl Storage {
         match incoming.message {
             NetRequest::Deploy(ref serialized_id) => {
                 let id = decode_item_id::<Deploy>(serialized_id)?;
-                let opt_item = self.get_deploy(id).map_err(FatalStorageError::from)?;
+                let mut opt_item = self.get_deploy(id).map_err(FatalStorageError::from)?;
+
+                match &mut opt_item {
+                    Some(deploy) => {
+                        let mut txn = self
+                            .env
+                            .begin_ro_txn()
+                            .expect("could not create RO transaction");
+                        if let Some(finalized_approvals) = txn
+                            .get_value(self.finalized_approvals_db, &id)
+                            .map_err(|e| {
+                                GetRequestError::Fatal(FatalStorageError::InternalStorage(e))
+                            })?
+                        {
+                            deploy.replace_approvals(finalized_approvals);
```

Comment:
So here, we would replace approvals we got from the GlobalState with approvals that the node has finalized? If they're different then:
*  Either our state root hashes are different (different approvals could lead to different execution result) and we will not sync up completely.
* Or it is a bug where the same set of deploy approvals (signatures) lead to the same execution results.


If it's the former then it's fine b/c joining node will not become a part of the same network. If it's the latter then I think we should actually use the signatures from the GlobalState. Otherwise users may find it confusing where the signatures from `finalized_approvals_db` differ from the ones in the GlobalState. The data should be coherent.

Reply:
I'm not sure I understand this -- I thought the approval sets could be different and were not previously stored in global state but in a table meant for them. Now, we're only storing the finalized approvals in GlobalState so no one's execution results should differ cause that must be the same after consensus.

Reply:
It's my fault. I misunderstood the branch we're here in.